File: src/main/java/lukas/week06/day4/Rehearsal.java
package lukas.week06.day4;

import java.util.Random;

public class Rehearsal {
    public static void main(String[] args) {
        int[] arr = new int[1000000];
        Random random = new Random();
        for (int i = 0; i < arr.length; i++) {
            arr[i] = random.nextInt();
        }

        ArrayStatistics statistics = getMinMaxAvg(arr);

        System.out.println(statistics.getMin());
        System.out.println(statistics.getMax());
        System.out.println(statistics.getAvg());

        System.out.println(statistics.toString());

        Object tmp = statistics;
        System.out.println(tmp.toString());
    }

    private static ArrayStatistics getMinMaxAvg(int[] arr) {
        int min = arr[0];
        int max = arr[0];
        double sum = 0;
        for (int value : arr) {
//            if (min > value) {
//                min = value;
//            }
//            if (max < value) {
//                max = value;
//            }
            min = Math.min(min, value);
            max = Math.max(max, value);
            sum += value;
        }
        double avg = sum / arr.length;
        return new ArrayStatistics(max, min, avg);
    }
}

File: src/main/java/lukas/week06/day4/kotlin/Rehearsal.kt
package lukas.week06.day4.kotlin

import java.util.*
import kotlin.math.max
import kotlin.math.min

fun main() {
    val arr = IntArray(1000000)
    val random = Random()
    for (i in arr.indices) {
        arr[i] = random.nextInt()
    }
    val statistics = getMinMaxAvg(arr)
    println(statistics.min)
    println(statistics.max)
    println(statistics.max)
    println(statistics.toString())
}

private fun getMinMaxAvg(arr: IntArray): ArrayStatistics {
    var min = arr[0]
    var max = arr[0]
    var sum = 0.0
    for (value in arr) {
//            if (min > value) {
//                min = value;
//            }
//            if (max < value) {
//                max = value;
//            }
        min = min(min, value)
        max = max(max, value)
        sum += value.toDouble()
    }
    val avg = sum / arr.size
    return ArrayStatistics(max, min, avg)
}

class ArrayStatistics(val max: Int = 0, val min: Int = 0, val avg: Double = 0.0) {
    override fun toString(): String {
        return """
               Minimum: $min
               Maximum: $max
               Average: $avg
               ${super.toString()}
               """.trimIndent()
    }
}